File: src/main/java/org/example/kalkulationsprogramm/dto/DokumentLockDto.java
package org.example.kalkulationsprogramm.dto;

import java.time.LocalDateTime;

/**
 * Antwort-DTO fuer den Dokument-Lock-Endpoint.
 *
 * status:
 *   ACQUIRED        - Der Caller haelt jetzt das Lock und darf bearbeiten.
 *   LOCKED_BY_OTHER - Ein anderer User haelt das Lock noch.
 */
public record DokumentLockDto(
        String status,
        Long holderUserId,
        String holderDisplayName,
        LocalDateTime acquiredAt,
        LocalDateTime lastHeartbeatAt
) {
    public static final String ACQUIRED = "ACQUIRED";
    public static final String LOCKED_BY_OTHER = "LOCKED_BY_OTHER";
}

File: src/main/kotlin/org/example/kalkulationsprogramm/dto/DokumentLockDto.kt
package org.example.kalkulationsprogramm.dto

import java.time.LocalDateTime

data class DokumentLockDto(
    val status: String,
    val holderUserId: Long?,
    val holderDisplayName: String?,
    val acquiredAt: LocalDateTime?,
    val lastHeartbeatAt: LocalDateTime?
) {
    fun status(): String = status
    fun holderUserId(): Long? = holderUserId
    fun holderDisplayName(): String? = holderDisplayName
    fun acquiredAt(): LocalDateTime? = acquiredAt
    fun lastHeartbeatAt(): LocalDateTime? = lastHeartbeatAt

    companion object {
        const val ACQUIRED = "ACQUIRED"
        const val LOCKED_BY_OTHER = "LOCKED_BY_OTHER"
    }
}
